sw/host/opentitanlib/src/debug/openocd.rs

```diff
@@ -204,6 +205,26 @@ impl OpenOcd {
         self.send(cmd)?;
         self.recv()
     }
+
+    /// Load instruction register of a given tap.
+    pub fn irscan(&mut self, tap: &str, ir: u32) -> Result<()> {
+        let cmd = format!("irscan {} {:#x}", tap, ir);
+        let result = self.execute(&cmd)?;
+        ensure!(result.is_empty(), "unexpected response: '{result}'");
+        Ok(())
+    }
+
+    /// Load data register of a given tap and return the scan.
+    pub fn drscan<T: ParseInt + LowerHex>(
+        &mut self,
+        tap: &str,
+        numbits: u32,
+        data: T,
+    ) -> Result<T> {
+        let cmd = format!("drscan {} {} {:#x}", tap, numbits, data);
+        let result = self.execute(&cmd)?;
+        Ok(T::from_str_radix(&result, 16).map_err(|x| x.into())?)
```

Comment:
Nit: doesn't the `?` operator already run `Into::into` for the error type?

Reply:
This from_str_radix is defined in our custom `ParseInt` trait which has `type FromStrRadixErr: Into<ParseIntError>`. This can't go directly via `?` to the anyhow error type since it would require two `.into()` calls.